### DISCUSSION TITLE
Booleans for selectionKeyCode

### DISCUSSION THREAD

**Discussion by moklick** [xyflow/xyflow#2167] [New Features]
**Update:** this is possible since v11.4 https://reactflow.dev/docs/guides/panning-and-zooming/#figma-like-viewport-controls

Currently it's not possible to create a figma/sketch like navigation experience:

- [x] create a selection by dragging the pane (no pressed key needed)
- [x] pan on scroll
- [x] pan on space and drag

We want to change that by making it possible to pass a boolean for the `selectionKeyCode` or add a new prop to handle this scenario.

**Comment by zyh374892663** [comment]
@moklick 
This is my own implementation of left-click direct box selection. If you have time, you can help to verify it to see if there are other problems. Looking forward to bringing you some inspiration and hel
[react-flow-renderer+10.3.7.patch.zip](https://github.com/wbkd/react-flow/files/8948480/react-flow-renderer%2B10.3.7.patch.zip)
p

**Reply by zyh374892663** [reply]
set `selectionKeyCode` to `null`

**Reply by moklick** [reply]
thanks @zyh374892663 I will have a look!

**Comment by zyh374892663** [comment]
@moklick
The previous patch did not take into account `panOnDrag`, support after the fix.
[react-flow-renderer+10.3.7.patch.zip](https://github.com/wbkd/react-flow/files/8954339/react-flow-renderer%2B10.3.7.patch.zip)

**Comment by reillyjodonnell** [comment]
Brainstorming here: selectOnDrag could be the boolean prop to enable this to work. Which obviously would default to false since the selectionKeyCode defaults to the string "Shift".

**Comment by antl3x** [comment]
news here? i'm about to use `react-selecto` to implement this behaviour. :(

**Reply by moklick** [reply]
This will be part of the next release https://github.com/wbkd/react-flow/pull/2661
